File: biogas_app/public/js/report/daily_slurry_summary/daily_slurry_summary.js
frappe.query_reports["Slurry Log Summary"] = {
    "filters": [
        {
            fieldname: "plant",
            label: __("Plant"),
            fieldtype: "Link",
            options: "Plant",
            reqd: 1
        },
        {
            fieldname: "from_date",
            label: __("From Date"),
            fieldtype: "Date",
            default: frappe.datetime.add_days(frappe.datetime.get_today(), -7),
            reqd: 1
        },
        {
            fieldname: "to_date",
            label: __("To Date"),
            fieldtype: "Date",
            default: frappe.datetime.get_today(),
            reqd: 1
        }
    ]
};

File: biogas_app/public/js/report/daily_slurry_summary/daily_slurry_summary.py
from __future__ import unicode_literals
import frappe
from frappe.utils import flt

def execute(filters=None):
    columns = [
        {"label": "Date", "fieldname": "date", "fieldtype": "Date", "width": 120},
        {"label": "Plant", "fieldname": "plant", "fieldtype": "Link", "options": "Plant", "width": 180},
        {"label": "Gas Output", "fieldname": "gas_output", "fieldtype": "Float", "width": 120},
    ]

    data = frappe.db.get_all(
        "Gas Output Log",
        filters={
            "plant": filters.get("plant"),
            "date": ["between", [filters.get("from_date"), filters.get("to_date")]]
        },
        fields=["date", "plant", "gas_output"]
    )

    return columns, data
